runtime/src/did.rs

```diff
@@ -296,8 +299,17 @@ decl_module! {
         ///
         /// [statechange]: ../enum.StateChange.html
         /// [keyupdate]: ./struct.KeyUpdate.html
-        // TODO: Use correct weight offset by benchmarking
-        #[weight = T::DbWeight::get().reads_writes(1, 1)  + 0]
+        /// # <weight>
+        /// This call requires a signature verification and the cost of verification varies by type
+        /// of signature
+        /// # </weight>
+        #[weight = T::DbWeight::get().reads_writes(1, 1) + {
+            match signature {
```

Comment:
Logic is repeated and numeric values are not equal to those within weight calculations for the `revoke` module: `(sr * 112_000_000) + (ed * 122_000_000) + (secp * 363_000_000)`.

Suggest implementing a new function:

```rust
impl DidSignature {
    /// Returns an upper bound on the resources required to verify this signature.
    /// The return value factors in both the computational cost to verify the signatures, and
    /// the required database reads.
    fn weight_to_verify(pauth: &self, dbw: &frame_support::weights::RuntimeDbWeight) -> u64;
}
```

Reply:
Good point. Will change.

Reply:
Done